src/test/kotlin/com/terraformation/backend/tracking/model/PlantingSiteBuilder.kt

```diff
@@ -0,0 +1,270 @@
+package com.terraformation.backend.tracking.model
+
+import com.terraformation.backend.db.SRID
+import com.terraformation.backend.db.default_schema.OrganizationId
+import com.terraformation.backend.db.tracking.MonitoringPlotId
+import com.terraformation.backend.db.tracking.PlantingSiteId
+import com.terraformation.backend.db.tracking.PlantingSubzoneId
+import com.terraformation.backend.db.tracking.PlantingZoneId
+import com.terraformation.backend.rectangle
+import com.terraformation.backend.rectanglePolygon
+import com.terraformation.backend.util.calculateAreaHectares
+import org.locationtech.jts.geom.GeometryFactory
+import org.locationtech.jts.geom.MultiPolygon
+import org.locationtech.jts.geom.PrecisionModel
+
+/**
+ * DSL for building planting site models with simple maps to test operations on site/zone/subzone
+ * geometry. The different layers of the map are all rectangular and are laid out horizontally
+ * starting at the southwest corner of the parent area.
+ *
+ * The DSL aims for succinctness and will try to build a simple but complete site if the caller
+ * doesn't specify otherwise.
+ *
+ * By default, each layer fills the remaining area of its parent. If a site has no zones, one is
```

Comment:
I think client might also be doing this.  Would be good to double-check to see if we have redundancy @tommylau523

Reply:
Not sure I understand the comment: are you saying we should make sure the client does this too, or that because the client does it, we don't need it in the code in this PR?

Reply:
Sorry I should've been more thorough in my response.  I think the client is already doing this (or will be once our user-drawn planting site feature is merged) because the UI prompts users for the "simple" approach or the "detailed" approach to drawing, which determines whether the user is asked to draw zones/subzones or not.  Obviously this doesn't cover shapefile uploads though.

Mainly I was pointing it out in case we wanted to remove the code from either client or server.

Reply:
Got it. I think there isn't any actual redundancy because the code in this PR is part of the test suite. There's no client code involved in the tests, and the code in this PR isn't included in the server.

Reply:
Ah, my misunderstanding then.  Should've read more carefully.